internal/backends/selector/lib.rs

```diff
@@ -46,6 +51,29 @@ cfg_if::cfg_if! {
             feature = "i-slint-backend-winit",
             feature = "i-slint-backend-linuxkms"
         ))] {
+        fn create_default_backend() -> Result<Box<dyn Platform + 'static>, PlatformError> {
+            let backends = [
+                #[cfg(all(feature = "i-slint-backend-qt", not(no_qt)))]
+                create_qt_backend,
+                #[cfg(feature = "i-slint-backend-winit")]
+                create_winit_backend,
+                #[cfg(feature = "i-slint-backend-linuxkms")]
+                create_linuxkms_backend,
+                || Err(PlatformError::NoPlatform),
+            ];
+
+            let mut last_err = PlatformError::NoPlatform;
+
+            for backend_factory in backends {
+                match backend_factory() {
+                    Ok(platform) => return Ok(platform),
+                    Err(err) => last_err = err,
```

Comment:
IMHO, this should not return just the last error, but a concatenation of all error. The error should be something like

```
Could not initialize backend.
Error from Qt backend: "Could not connect to display server"
Error from Winit backend: XOpenDisplayFailed
Error from LinuxKMS: could not dlopen shared library "libseat"
```

You get the idea.

Reply:
Okay, good point. I can do that.

Reply:
I implemented this now, but it's... sub-optimal:

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Other("Could not initialize backend.\nError from Winit backend: Error initializing winit event loop: os error at /home/slint/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.29.2/src/platform_impl/linux/mod.rs:776: neither WAYLAND_DISPLAY nor DISPLAY is set.\nError from LinuxKMS backend: Timeout while waiting to activate session\nNo backends configured.")', examples/carousel/rust/main.rs:23:23
```

Should I introduce a dedicated PlatformError instead that holds a Vec of other Errors? Or is it perhaps wrong to return an Error here anyway and we should panic?

Reply:
I think it's good for now.

We may want to improve the Debug for our error maybe.
But that can be done another time.